client/models.js

```diff
@@ -12,28 +13,38 @@ export class Bank {
 }
 
 export class Account {
-    constructor(arg) {
-        this.bank          = has(arg, 'bank') && arg.bank;
-        this.bankAccess    = has(arg, 'bankAccess') && arg.bankAccess;
-        this.title         = has(arg, 'title') && arg.title;
-        this.accountNumber = has(arg, 'accountNumber') && arg.accountNumber;
-        this.initialAmount = has(arg, 'initialAmount') && arg.initialAmount;
-        this.lastChecked   = has(arg, 'lastChecked') && new Date(arg.lastChecked);
-        this.id            = has(arg, 'id') && arg.id;
-        this.iban          = (maybeHas(arg, 'iban') && arg.iban) || null;
+    constructor(arg, defaultCurrency) {
+        this.bank              = has(arg, 'bank') && arg.bank;
+        this.bankAccess        = has(arg, 'bankAccess') && arg.bankAccess;
+        this.title             = has(arg, 'title') && arg.title;
+        this.accountNumber     = has(arg, 'accountNumber') && arg.accountNumber;
+        this.initialAmount     = has(arg, 'initialAmount') && arg.initialAmount;
+        this.lastChecked       = has(arg, 'lastChecked') && new Date(arg.lastChecked);
+        this.id                = has(arg, 'id') && arg.id;
+        this.iban              = (maybeHas(arg, 'iban') && arg.iban) || null;
+        // By default, the currency is Euro.
+
+        this.currency          = (maybeHas(arg, 'currency') && currency.isKnown(arg.currency) &&
+                                  arg.currency) || defaultCurrency;
```

Comment:
Can't the initial condition fit on one line? If not, can you instead put it as one part on each line?

```
(A &&
B &&
C) ||
D;
```